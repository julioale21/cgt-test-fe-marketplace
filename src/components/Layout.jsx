import { Outlet } from 'react-router-dom';
import { Header } from '../components/shared/Header';
import { Box } from '@mui/material';

export function Layout() {
  const cartItemsCount = 3;

  return (
    <Box sx={{ display: 'flex', flexDirection: 'column', minHeight: '100vh' }}>
      <Header cartItemsCount={cartItemsCount} />

      <Box component="main" sx={{ flexGrow: 1, p: 3 }}>
        <Outlet />
      </Box>

      <Box component="footer" sx={{ p: 2, textAlign: 'center', mt: 'auto' }}>
        <p>Â© 2025</p>
      </Box>
    </Box>
  );
}
