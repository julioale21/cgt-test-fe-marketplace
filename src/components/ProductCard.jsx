import React from 'react';

import { Link } from 'react-router-dom';

import { Button, Card, CardActions, CardContent, CardMedia, Grid, Typography } from '@mui/material';
import ShoppingCartIcon from '@mui/icons-material/ShoppingCart';

const ProductCard = ({ product }) => {
  return (
    <Grid item xs={12} sm={6} md={4} lg={3}>
      <Card sx={{ height: '100%', display: 'flex', flexDirection: 'column' }}>
        <CardMedia
          component="img"
          height="200"
          image={product.image}
          alt={product.name}
          sx={{
            objectFit: 'cover',
            height: '200px'
          }}
        />
        <CardContent sx={{ flexGrow: 1 }}>
          <Typography gutterBottom variant="h6" component="h2">
            {product.name}
          </Typography>
          <Typography variant="body2" color="text.secondary" sx={{ mb: 2 }}>
            {product.description}
          </Typography>
          <Typography variant="h6" color="primary" sx={{ mt: 2 }}>
            ${product.price.toFixed(2)}
          </Typography>
          <Typography variant="body2" color="text.secondary" sx={{ mt: 1 }}>
            Formats: {product.format.join(', ')}
          </Typography>
        </CardContent>
        <CardActions sx={{ justifyContent: 'space-between' }}>
          <Button component={Link} to={`/products/${product.id}`} size="small" color="primary">
            View Details
          </Button>
          <Button size="small" variant="contained" color="primary" startIcon={<ShoppingCartIcon />}>
            Add to Cart
          </Button>
        </CardActions>
      </Card>
    </Grid>
  );
};

export { ProductCard };
