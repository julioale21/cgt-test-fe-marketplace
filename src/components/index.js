export { ContainedButton } from "./buttons/ContainedButton";
export { Header } from "./shared/Header";
export { Layout } from "./Layout";
export { OutlinedButton } from "./buttons/OutlinedButton";
export { ProductList } from "./ProductList";
export {ProductCard} from "./ProductCard";
