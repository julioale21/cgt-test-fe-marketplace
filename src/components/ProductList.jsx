import React from 'react';
import { Grid } from '@mui/material';
import { ProductCard } from './ProductCard';

const ProductList = ({ products }) => {
  return (
    <Grid container spacing={4}>
      {products.map((product) => (
        <ProductCard key={product.id} product={product} />
      ))}
    </Grid>
  );
};

export { ProductList };
