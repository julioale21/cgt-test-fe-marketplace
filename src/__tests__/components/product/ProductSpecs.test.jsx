import { render, screen } from '@testing-library/react';
import { ProductSpecs } from '../../../components';

describe('ProductSpecs', () => {
  const mockProps = {
    format: ['FBX', 'OBJ', 'MAX'],
    polygons: '24k',
    textures: '4K PBR'
  };

  it('should match snapshot', () => {
    const props = { ...mockProps };

    const { container } = render(<ProductSpecs {...props} />);

    expect(container).toMatchSnapshot();
  });

  it('should render all specification labels', () => {
    const props = { ...mockProps };

    render(<ProductSpecs {...props} />);

    expect(screen.getByText('Available Formats')).toBeInTheDocument();
    expect(screen.getByText('Polygons')).toBeInTheDocument();
    expect(screen.getByText('Textures')).toBeInTheDocument();
  });

  it('should display formats as comma-separated list', () => {
    const props = { ...mockProps };

    render(<ProductSpecs {...props} />);

    expect(screen.getByText(props.format.join(', '))).toBeInTheDocument();
  });

  it('should show correct polygons value', () => {
    const props = { ...mockProps };

    render(<ProductSpecs {...props} />);

    expect(screen.getByText(props.polygons)).toBeInTheDocument();
  });

  it('should display textures information', () => {
    const props = { ...mockProps };

    render(<ProductSpecs {...props} />);

    expect(screen.getByText(props.textures)).toBeInTheDocument();
  });

  it('should render dividers between specs', () => {
    const props = { ...mockProps };

    const { container } = render(<ProductSpecs {...props} />);

    const dividers = container.querySelectorAll('.MuiDivider-root');
    expect(dividers).toHaveLength(2);
  });

  it('should throw error when required props are missing', () => {
    const consoleSpy = jest.spyOn(console, 'error').mockImplementation(() => {});

    render(<ProductSpecs />);

    expect(consoleSpy).toHaveBeenCalled();
    consoleSpy.mockRestore();
  });
});
