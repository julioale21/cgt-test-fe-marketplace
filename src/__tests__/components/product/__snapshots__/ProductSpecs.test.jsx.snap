// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductSpecs should match snapshot 1`] = `
<div>
  <ul
    class="MuiList-root MuiList-padding css-b7dj7n-MuiList-root"
  >
    <li
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-cfd5wn-MuiListItem-root"
    >
      <div
        class="MuiListItemText-root MuiListItemText-multiline css-14ln1j6-MuiListItemText-root"
      >
        <span
          class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-rizt0-MuiTypography-root"
        >
          Available Formats
        </span>
        <p
          class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-1a1whku-MuiTypography-root"
        >
          FBX, OBJ, MAX
        </p>
      </div>
    </li>
    <hr
      class="MuiDivider-root MuiDivider-fullWidth css-1jpc804-MuiDivider-root"
    />
    <li
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-cfd5wn-MuiListItem-root"
    >
      <div
        class="MuiListItemText-root MuiListItemText-multiline css-14ln1j6-MuiListItemText-root"
      >
        <span
          class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-rizt0-MuiTypography-root"
        >
          Polygons
        </span>
        <p
          class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-1a1whku-MuiTypography-root"
        >
          24k
        </p>
      </div>
    </li>
    <hr
      class="MuiDivider-root MuiDivider-fullWidth css-1jpc804-MuiDivider-root"
    />
    <li
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-cfd5wn-MuiListItem-root"
    >
      <div
        class="MuiListItemText-root MuiListItemText-multiline css-14ln1j6-MuiListItemText-root"
      >
        <span
          class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-rizt0-MuiTypography-root"
        >
          Textures
        </span>
        <p
          class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-1a1whku-MuiTypography-root"
        >
          4K PBR
        </p>
      </div>
    </li>
  </ul>
</div>
`;
